server/socket.c

```diff
@@ -21,16 +21,26 @@
 #include <arpa/inet.h>
 #include <signal.h>
 #include <netinet/in.h>
-
+#include <pthread.h>
 
 #include <arpa/inet.h>
+
+/**********DEFINES**********/
+#define REC_LEN 128
+#define FILE_DATA_LEN 1024
 /***********GLOBAL_VARIBLES*************/
 struct addrinfo hints;
 struct addrinfo* servinfo;
 struct sockaddr* socketaddr;
 int sock_fd;
 const char* file_aesdsocket = "/var/tmp/aesdsocketdata";
+int accepted = 0; //data_sent_flag = 0;
 
+struct thread_data{
+	pthread_t thread_use;
+    uint32_t thread_id;
+    struct thread_data nxt_thread;
+};
 static void signal_handler(int signo)
 {
     if(signo == SIGINT)
```

Comment:
This is not re-entrant. Ensure to just set a flag and clean up in the main thread context.